src/jv_aux.c

```diff
@@ -222,13 +222,17 @@ jv jv_dels(jv t, jv keys) {
     // no change
   } else if (jv_get_kind(t) == JV_KIND_ARRAY) {
     // extract slices, they must be handled differently
-    jv orig_keys = keys;
-    keys = jv_array();
+    jv neg_keys = jv_array();
+    jv nonneg_keys = jv_array();
     jv new_array = jv_array();
     jv starts = jv_array(), ends = jv_array();
-    jv_array_foreach(orig_keys, i, key) {
+    jv_array_foreach(keys, i, key) {
       if (jv_get_kind(key) == JV_KIND_NUMBER) {
-        keys = jv_array_append(keys, key);
+        if (jv_number_value(key) < 0) {
+          neg_keys = jv_array_append(neg_keys, key);
+        } else {
+          nonneg_keys = jv_array_append(nonneg_keys, key);
```

Comment:
No because the code below expects indices to be sorted. Say you start with an array of size 10 and indices [-2, -1, 1, 2]. After normalizing, it would be [8, 9, 1, 2]. Solution A would be to rotate the array after this loop to make it sorted again, which I am not happy with. Solution B would be to keep track of the first nonnegative index while normalizing, then continue to use two pointers in the code below, but they would be pointers into the same array. The code for solution B would look roughly the same as this solution, but I think having separate arrays is simpler to understand.